chapters/VK_KHR_swapchain/wsi.txt

```diff
@@ -258,7 +258,7 @@ flink:vkQueuePresentKHR any images it had already acquired from
 pname:oldSwapchain.
 E.g., an application may present an image from the old swapchain before an
 image from the new swapchain is ready to be presented.
-As usual, flink:vkQueuePresentKHR may: fail if pname:oldSwapchain has
+flink:vkQueuePresentKHR must: fail if pname:oldSwapchain has
 entered a state that causes ename:VK_ERROR_OUT_OF_DATE_KHR to be returned.
```

Comment:
This does not fix the full extent of the complaint in the original Issue.

The problem is also the sentence is tautological. If `VK_ERROR_OUT_OF_DATE_KHR` was returned, then by definition it did fail (no may or must about it).

I think the intention of the sentence is following  (and so should the wording match that)

> _Retired_ `oldSwapchain` may: still become unusable as much as non-retired swapchain. When that happens,  it is treated identically by `vkQueuePresentKHR`; the command **must** return `VK_ERROR_OUT_OF_DATE_KHR`.

Reply:
Reworded.